### ISSUE TITLE
Accept `Stream` for batch query execution

### ISSUE THREAD

**Issue by chilloutman** [jOOQ/jOOQ#6500]
Currently [`DSLContext.batch`](https://github.com/jOOQ/jOOQ/blob/63f3278d777dfd660f5d3a6b81857ee9f79ec6d5/jOOQ/src/main/java/org/jooq/impl/DefaultDSLContext.java#L2693) and it's variants like `batchStore`, `batchInsert`, etc. accept arguments of type `Collection`. How about adding `java.util.Stream` variants?

This is just a QoL improvement, as one can always `collect(toList())` and use the existing methods.

---

Here's an example use-case:

```java
Stream<Records> records = domainObjects
  .stream()
  .map(this::toRecord);

create.batchStore(records);
```

Similar to how the methods accepting `Collection` [map the collection to an array](https://github.com/jOOQ/jOOQ/blob/63f3278d777dfd660f5d3a6b81857ee9f79ec6d5/jOOQ/src/main/java/org/jooq/impl/DefaultDSLContext.java#L2719), the implementation for `Stream` could look like this:

```java
public Batch batch(Stream<? extends Query> queries) {
    return batch(queries.toArray(Query[]::new));
}
```

**Comment by lukaseder** [comment]
Thanks for your suggestion. You're absolutely right, this would be a great improvement in the future, not just for batching, but for pretty much any API element that currently accepts arrays or collections. We have a pending feature request for this. I've rescheduled it for 3.11 - it won't fit the 3.10 schedule anymore:
https://github.com/jOOQ/jOOQ/issues/5133

Thanks again.

**Comment by lukaseder** [comment]
Closed as duplicate of #5133
